File: CS/WebSite/App_Code/CommandColumnHeaderTemplate.cs
using System.Web.UI;
using System.Web.UI.HtmlControls;
using DevExpress.Web.ASPxGridView;

public class CommandColumnHeaderTemplate : ITemplate {
    private ASPxGridView gridView = null;
    
    public CommandColumnHeaderTemplate(ASPxGridView gridView) {
        this.gridView = gridView;
    }

    #region ITemplate Members

    public void InstantiateIn(Control container) {
        HtmlInputCheckBox checkBox = new HtmlInputCheckBox();
        container.Controls.Add(checkBox);
        checkBox.Attributes.Add("onclick", string.Format("{0}.SelectAllRowsOnPage(this.checked);", gridView.ClientInstanceName));
    }

    #endregion
}

File: VB/WebSite/App_Code/CommandColumnHeaderTemplate.vb
Imports System.Web.UI
Imports System.Web.UI.HtmlControls
Imports DevExpress.Web.ASPxGridView

Public Class CommandColumnHeaderTemplate
    Implements ITemplate

    Private gridView As ASPxGridView = Nothing

    Public Sub New(ByVal gridView As ASPxGridView)
        Me.gridView = gridView
    End Sub

    #Region "ITemplate Members"

    Public Sub InstantiateIn(ByVal container As Control) Implements ITemplate.InstantiateIn
        Dim checkBox As New HtmlInputCheckBox()
        container.Controls.Add(checkBox)
        checkBox.Attributes.Add("onclick", String.Format("{0}.SelectAllRowsOnPage(this.checked);", gridView.ClientInstanceName))
    End Sub

    #End Region
End Class
